website/pages/docs/theming/scale-out-theme.mdx

```diff
@@ -0,0 +1,519 @@
+  headerNavigation: 1750,
+  // More...
+}
+
+export type ZIndices = typeof zIndices
+
+export default zIndices
+```
+
+### 📦 Export Foundations
+Export all above overwritten Foundations in `index.ts`. Put this file under `/foundations` folder.
+
+```ts live=false
+// index.ts
+
+import borders from "./borders"
+import breakpoints from "./breakpoints"
+import colors from "./colors"
+import radii from "./radius"
+import shadows from "./shadows"
+import sizes from "./sizes"
+import { spacing } from "./spacing"
+import transition from "./transition"
+import typography from "./typography"
+import zIndices from "./z-index"
+
+const foundations = {
+  breakpoints,
+  zIndices,
+  radii,
+  colors,
+  ...typography,
+  sizes,
+  shadows,
+  space: spacing,
+  borders,
+  transition,
+}
+
+type FoundationsType = typeof foundations
+export interface Foundations extends FoundationsType {}
+
+export default foundations
+```
+
+---
+
+## 📁 Components
+
+<ComponentLinks
+  github={{ package: "theme/src/components/" }}
+/>
+
+In this folder, we define the style that will be the default for the components.
+
+We will learn [how to define the Component Styles in the next section](/docs/theming/component-style).
+For that reason, we only provides a simple example for splitting Component Styles into files.
+
+
+### 🛎 Button Component Styles
+We can define `<Button>` component styles like this:
+
+```ts live=false
+// button.ts
+
+// Styles for the base style
+const baseStyle = {
+  fontWeight: 'bold',
+  textTransform: 'uppercase',
+}
+
+// Styles for the size variations
+const sizes = {
+  sm: {
+    fontSize: '12px',
+    padding: '16px',
+  },
+  md: {
+    fontSize: '16px',
+    padding: '24px',
+  },
+}
+
+// Styles for the visual style variations
+const variants = {
+  outline: {
+    border: '2px solid',
+    borderColor: 'green.500',
+  },
+  solid: {
+    bg: 'green.500',
+    color: 'white',
+  },
+}
+
+// The default `size` or `variant` values
+const defaultProps = {
+  size: 'md',
+  variant: 'outline',
+}
+
+export default {
+  baseStyle,
+  sizes,
+  variants,
+  defaultProps,
+}
+```
+
+
+### 📦 Export Component Styles
+Export all Component Styles in `index.ts`. Put this file under `/components` folder.
+
+```ts live=false
+// index.ts
+
+import Button from './button';
+// More...
+
+export default {
+  Button,
+  // More...
+}
+```
+
+---
+
+## 📦 Export Theme
+Finally, we will merge all the overwrites and export the theme.
+Put `index.ts` directly under `/theme` folder.
+
+You can learn more about [ColorModeOptions](/docs/features/color-mode) and [ThemeDirection](/docs/features/rtl-support).
+
+```ts live=false
+// index.ts
+
+import { extendTheme } from '@chakra-ui/react';
+import { ColorModeOptions } from '@chakra-ui/system';
+import styles from './styles';
+import foundations from './foundations';
+import components from './components';
+
+export type ThemeConfig = ColorModeOptions;
+
+export type ThemeDirection = 'ltr' | 'rtl';
```

Comment:
Please remove every additional TypeScript types. Projects in userland should import the types from @chakra-ui/react, and not create their own shadow type. This helps maintaining the theme over the next chakra versions.